registry.go

```diff
@@ -104,6 +105,11 @@ type Layer interface {
 
 	// CreatedAt returns the time this layer was created.
 	CreatedAt() time.Time
+
+	// ServeHTTP allows a layer to serve itself, whether by providing
+	// a redirect directly to the content, or by serving the content
+	// itself
+	ServeHTTP(w http.ResponseWriter, r *http.Request)
```

Comment:
@endophage The other option is to have callers try a type assertion for `(http.Handler).ServeHTTP`. Wrappers would have to be careful to forward this through when wrapping `Blob`.